planet/reporting.py

```diff
@@ -90,15 +90,19 @@ def update_state(self, state: str):
     def update(self, state: str = None, order_id: str = None):
         if state:
             self.state = state
-            try:
-                self.bar.postfix[1] = self.state
-            except AttributeError:
-                # If the bar is disabled, attempting to access self.bar.postfix
-                # will result in an error. In this case, just skip it.
-                pass
+            if self.bar is not None:
+                try:
+                    self.bar.postfix[1] = self.state
+                except AttributeError:
+                    # If the bar is disabled, attempting to access
+                    # self.bar.postfix will result in an error. In this
+                    # case, just skip it.
+                    pass
 
         if order_id:
             self.order_id = order_id
-            self.bar.set_description_str(self.desc, refresh=False)
+            if self.bar is not None:
+                self.bar.set_description_str(self.desc, refresh=False)
 
-        self.bar.refresh()
+        if self.bar is not None:
+            self.bar.refresh()
```

Comment:
mypy noticed some cases where self.bar could be None and we would try to call its methods.

Reply:
That is cool as heck. I should try mypy

Reply:
awesome catch. this would probably have bubbled up as a bug eventually.